src-web/components/modals/BulkPolicyActionModal.js

```diff
@@ -0,0 +1,325 @@
+		this.handleCloseClick = this.handleCloseClick.bind(this)
+	}
+
+	handleSubmitClick() {
+		const {
+			type: modalType,
+			handleSubmit,
+			handleDeleteSubmit,
+			data = [],
+		} = this.props
+		switch (modalType) {
+			case 'bulk-policy-action-disable':
+				handleSubmit(
+					data,
+					true,
+					'/spec/disabled',
+					modalType,
+					bulkPolicyActions
+				)
+				break
+			case 'bulk-policy-action-enable':
+				handleSubmit(
+					data,
+					false,
+					'/spec/disabled',
+					modalType,
+					bulkPolicyActions
+				)
+				break
+			case 'bulk-policy-action-enforce':
+				handleSubmit(
+					data,
+					'enforce',
+					'/spec/remediationAction',
+					modalType,
+					bulkPolicyActions
+				)
+				break
+			case 'bulk-policy-action-inform':
+				handleSubmit(
+					data,
+					'inform',
+					'/spec/remediationAction',
+					modalType,
+					bulkPolicyActions
+				)
+				break
+			case 'bulk-policy-action-delete':
+				handleDeleteSubmit(data, modalType, bulkRemovePolicies)
+				break
+			default:
+				break
+		}
+	}
+
+	handleCloseClick() {
+		const { type: modalType, handleClose } = this.props
+		handleClose(modalType)
+	}
+
+	render() {
+		const {
+			actionType,
+			type: modalType,
+			data,
+			label,
+			locale,
+			open,
+			reqErrorMsg,
+			reqStatus,
+		} = this.props
+		return (
+			<div>
+				{reqStatus === REQUEST_STATUS.IN_PROGRESS && (
+					<Spinner className="patternfly-spinner" />
+				)}
+				<AcmModal
+					titleIconVariant={
+						modalType === 'bulk-policy-action-delete'
+							? 'warning'
+							: null
+					}
+					variant="medium"
+					id={`bulk-${actionType}-policy-modal'`}
+					isOpen={open}
+					showClose={true}
+					onClose={this.handleCloseClick}
+					title={
+						data.length > 1
+							? msgs.get(label.heading, [data.length], locale)
+							: msgs.get(
+									`modal.${actionType}-hcmcompliance.heading`,
+									locale
+							)
+					}
+					actions={[
+						<AcmButton
+							key="confirm"
+							variant={
+								modalType === 'bulk-policy-action-delete'
+									? ButtonVariant.danger
+									: ButtonVariant.primary
+							}
+							onClick={this.handleSubmitClick}
+						>
+							{msgs.get(label.primaryBtn, locale)}
+						</AcmButton>,
+						<AcmButton
+							key="cancel"
+							variant={ButtonVariant.link}
+							onClick={this.handleCloseClick}
+						>
+							{msgs.get('modal.button.cancel', locale)}
+						</AcmButton>,
+					]}
+				>
+					{reqStatus === REQUEST_STATUS.ERROR && (
+						<AcmAlert
+							isInline={true}
+							noClose={true}
+							variant="warning"
+							title={
+								reqErrorMsg ||
+								msgs.get('error.default.description', locale)
+							}
+						/>
+					)}
+					<p style={{ marginBottom: '.5rem' }}>
+						{msgs.get(
+							`modal.actions.bulk.${actionType}.description`,
+							locale
+						)}
+					</p>
+					{data.some(
+						(policy) => policy.source.text.toLowerCase() !== 'local'
```

Comment:
I believe the text is set up for translation, so this could break in the future when we get translated.

Reply:
Approving this PR though - it looks great, and we can handle that in a subsequent one.